File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/brent/meduza/MeduzaStockProvider.java
package com.github.bumblebee.command.brent.meduza;

import com.github.bumblebee.command.brent.meduza.response.MeduzaStockResponse;
import feign.Client;
import feign.Feign;
import feign.gson.GsonDecoder;
import feign.slf4j.Slf4jLogger;
import org.springframework.stereotype.Service;

import javax.net.ssl.SSLContext;
import javax.net.ssl.SSLSocketFactory;
import javax.net.ssl.TrustManager;
import javax.net.ssl.X509TrustManager;
import javax.security.cert.X509Certificate;
import java.security.KeyManagementException;
import java.security.NoSuchAlgorithmException;
import java.security.cert.CertificateException;

@Service
public class MeduzaStockProvider {

    private final MeduzaApi client;

    public MeduzaStockProvider() {

        this.client = Feign.builder()
                .decoder(new GsonDecoder())
                .logLevel(feign.Logger.Level.BASIC)
                .logger(new Slf4jLogger())
                .client(new Client.Default(allTrustSocketFactory(), (hostname, session) -> true))
                .target(MeduzaApi.class, MeduzaApi.API_ROOT);
    }

    SSLSocketFactory allTrustSocketFactory() {

        // Create a trust manager that does not validate certificate chains
        TrustManager[] trustAllCerts = new TrustManager[]{
                new X509TrustManager() {
                    @Override
                    public void checkClientTrusted(java.security.cert.X509Certificate[] x509Certificates, String s) throws CertificateException {

                    }

                    @Override
                    public void checkServerTrusted(java.security.cert.X509Certificate[] x509Certificates, String s) throws CertificateException {

                    }

                    public java.security.cert.X509Certificate[] getAcceptedIssuers() {
                        return null;
                    }

                    public void checkClientTrusted(X509Certificate[] certs, String authType) {
                    }

                    public void checkServerTrusted(X509Certificate[] certs, String authType) {
                    }
                }
        };

        try {
            SSLContext sc = SSLContext.getInstance("SSL");
            sc.init(null, trustAllCerts, new java.security.SecureRandom());
            return sc.getSocketFactory();
        } catch (NoSuchAlgorithmException | KeyManagementException e) {
            throw new RuntimeException(e);
        }
    }

    public MeduzaStockResponse getCurrentStocks() {

        return this.client.queryStock();
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/brent/meduza/MeduzaStockProvider.kt
package com.github.bumblebee.command.brent.meduza

import feign.Client
import feign.Feign
import feign.gson.GsonDecoder
import feign.slf4j.Slf4jLogger
import org.springframework.stereotype.Service
import java.security.cert.CertificateException
import javax.net.ssl.SSLContext
import javax.net.ssl.SSLSocketFactory
import javax.net.ssl.TrustManager
import javax.net.ssl.X509TrustManager

@Service
class MeduzaStockProvider {

    private val client: MeduzaApi = Feign.builder()
            .decoder(GsonDecoder())
            .logLevel(feign.Logger.Level.BASIC)
            .logger(Slf4jLogger())
            .client(Client.Default(allTrustSocketFactory()) { _, _ -> true })
            .target(MeduzaApi::class.java, MeduzaApi.API_ROOT)

    fun getCurrentStocks(): MeduzaStockResponse = this.client.queryStock()

    private fun allTrustSocketFactory(): SSLSocketFactory {

        // Create a trust manager that does not validate certificate chains
        val trustAllCerts = arrayOf<TrustManager>(object : X509TrustManager {
            @Throws(CertificateException::class)
            override fun checkClientTrusted(x509Certificates: Array<java.security.cert.X509Certificate>, s: String) {

            }

            @Throws(CertificateException::class)
            override fun checkServerTrusted(x509Certificates: Array<java.security.cert.X509Certificate>, s: String) {

            }

            override fun getAcceptedIssuers(): Array<java.security.cert.X509Certificate>? {
                return null
            }
        })

        val sc = SSLContext.getInstance("SSL")
        sc.init(null, trustAllCerts, java.security.SecureRandom())
        return sc.socketFactory
    }
}
